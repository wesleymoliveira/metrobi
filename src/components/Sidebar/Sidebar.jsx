import { Typography } from '../Typography/Typography';
import * as S from './Sidebar.styled';

export function Sidebar() {
	return (
		<S.Wrapper data-testid='sidebar'>
			<Typography size='xxlarge'>
				Sidebar
			</Typography>
		</S.Wrapper>
	);
}
