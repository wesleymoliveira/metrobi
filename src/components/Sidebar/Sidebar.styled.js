import styled, { css } from 'styled-components'

export const Wrapper = styled.aside`
  ${({ theme }) => css`
    background-color: ${theme.colors.sidebarBackground};
    flex: 3;
    display: flex;
    justify-content: center;
    align-items: start;
  `}
`;
