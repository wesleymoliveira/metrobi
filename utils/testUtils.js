import React from "react";
import { ThemeProvider } from "styled-components";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";

import theme from "../src/styles/theme";

const AllTheProviders = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (ui, options) => {
  return render(ui, { wrapper: AllTheProviders, ...options });
};

const snapshotRenderer = (component) => {
  return renderer.create(
    <ThemeProvider theme={theme}>{component}</ThemeProvider>
  );
};

export * from "@testing-library/react";

export { customRender as render, snapshotRenderer  };
