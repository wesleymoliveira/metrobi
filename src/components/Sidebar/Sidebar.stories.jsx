import React from 'react';
import { Sidebar } from './Sidebar';

const SidebarArgTypes = {
  label: {
    name: 'label',
    type: { name: 'string', required: false },
    description: 'overwritten description',
    table: {
      type: { 
        summary: 'something short', 
        detail: 'something really really long' 
      },

    },
    control: {
      type: 'string'
    }
  }
};

export default {
	title: 'components/Sidebar',
	component: Sidebar,
	argTypes: SidebarArgTypes
};

export const Default = (args) => <Sidebar {...args} />;
