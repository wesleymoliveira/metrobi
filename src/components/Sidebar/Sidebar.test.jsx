import { screen } from '@testing-library/react';

import { Sidebar } from './Sidebar';
import { render, snapshotRenderer } from '../../../utils/testUtils';

describe('<Sidebar />', () => {
		it('should render the component', () => {
		render(<Sidebar />);
		
		expect(screen.getByTestId(/sidebar/i)).toBeInTheDocument();
	});

	it('should match snapshot', () => {
		const tree = snapshotRenderer(<Sidebar />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should match the original background-color', () => {
		const tree = snapshotRenderer(<Sidebar />).toJSON();
		//make sure the background-color is the original one
		expect(tree).toHaveStyleRule('background-color', '#9FC363')
	});

});
