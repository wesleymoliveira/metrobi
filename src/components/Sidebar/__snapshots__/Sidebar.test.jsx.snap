// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> should match snapshot 1`] = `
.c1 {
  color: #FAFAFA;
  font-size: 2.8rem;
}

.c0 {
  background-color: #9FC363;
  -webkit-flex: 3;
  -ms-flex: 3;
  flex: 3;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

<aside
  className="c0"
  data-testid="sidebar"
>
  <h1
    className="c1"
    color="white"
    data-testid="typography"
    size="xxlarge"
  >
    Sidebar
  </h1>
</aside>
`;
